### ISSUE TITLE
ExtractorError: YouTube said: Unable to extract video data

### ISSUE THREAD

**Issue by theocrite** [ytdl-org/youtube-dl#26152]
## Checklist

- [x] I'm reporting a broken site support
- [x] I've verified that I'm running youtube-dl version **2020.07.28**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar issues including closed ones

## Verbose log

youtube-dl -v P5NB1H5pCHk
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'-v', u'P5NB1H5pCHk']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2020.07.28
[debug] Python version 2.7.16 (CPython) - Linux-4.19.0-9-amd64-x86_64-with-debian-10.4
[debug] exe versions: ffmpeg 4.1.6-1, ffprobe 4.1.6-1, phantomjs 2.1.1, rtmpdump 2.4
[debug] Proxy map: {}
[youtube] P5NB1H5pCHk: Downloading webpage
ERROR: P5NB1H5pCHk: YouTube said: Unable to extract video data
Traceback (most recent call last):
  File "/home/theo/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 797, in extract_info
    ie_result = ie.extract(url)
  File "/home/theo/bin/youtube-dl/youtube_dl/extractor/common.py", line 530, in extract
    ie_result = self._real_extract(url)
  File "/home/theo/bin/youtube-dl/youtube_dl/extractor/youtube.py", line 1893, in _real_extract
    'YouTube said: %s' % unavailable_message, expected=True, video_id=video_id)
ExtractorError: P5NB1H5pCHk: YouTube said: Unable to extract video data

youtube-dl -v 4wZPsFWfi3k
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'-v', u'4wZPsFWfi3k']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2020.07.28
[debug] Python version 2.7.16 (CPython) - Linux-4.19.0-9-amd64-x86_64-with-debian-10.4
[debug] exe versions: ffmpeg 4.1.6-1, ffprobe 4.1.6-1, phantomjs 2.1.1, rtmpdump 2.4
[debug] Proxy map: {}
[youtube] 4wZPsFWfi3k: Downloading webpage
ERROR: 4wZPsFWfi3k: YouTube said: Unable to extract video data
Traceback (most recent call last):
  File "/home/theo/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 797, in extract_info
    ie_result = ie.extract(url)
  File "/home/theo/bin/youtube-dl/youtube_dl/extractor/common.py", line 530, in extract
    ie_result = self._real_extract(url)
  File "/home/theo/bin/youtube-dl/youtube_dl/extractor/youtube.py", line 1893, in _real_extract
    'YouTube said: %s' % unavailable_message, expected=True, video_id=video_id)
ExtractorError: 4wZPsFWfi3k: YouTube said: Unable to extract video data

## Description

It seems that ytdl can't download restricted videos anymore (the ones behind a "sign in to confirm your age" wall).

**Comment by guestbuy5** [comment]
I'm seeing the same issue with latest version.

**Comment by oSumAtrIX** [comment]
I can confirm that this is also happening with with HW4IM87DAdg

**Comment by ethanbreck** [comment]
Seems like a mass issue, as issues #26148 #26123 #26103  is having very similar issues.

**Comment by jshir** [comment]
Both videos above worked for me - only after I fed it my YouTube cookies.

I came here after noticing a similar problem with _Uelrsz_VfY (even though video is not age restricted and plays in private mode just fine). It seems like YT is requiring a session cookie to dl certain videos (maybe to do with country restrictions as this video got song matched).

As more videos have this issue I think it would be a nice enhancement for YTDL to give the user some hand-holding as far as getting cookies.txt set up and automatically using it when needed instead of adding --cookies c:\path\to\youtube-dl\cookies.txt every time.

**Comment by ethanbreck** [comment]
Yeah, like a youtube-dl --first or youtube-dl --cookie-update that sort of walks you through it would be nice.

**Comment by Xan-Kun** [comment]
Hey there!
Same here PrJ5Sx1Dtc8 . So is the (missing) cookie actually the issue here?
Can someone confirm?

**Comment by bustbr** [comment]
PR #26100 fixes the age-gate check, but it's not merged into upstream yet. You can apply the patch locally.

**Comment by Xan-Kun** [comment]
I ask because I tried two approaches to provide cookies for a logged-in user and both didn't work (providing text file and --add-header option).
I assume I got something wrong in the format then...

**Comment by BrimDragster** [comment]
Happens with yJj9q-67ivA too.

**Comment by chudsaviet** [comment]
Another example with this error: https://www.youtube.com/watch?v=oL276M79mYE .

**Comment by mixon2020** [comment]
I am having the same issue as #26299.
here is what my software gave me as the issue
====================================
ERROR: Unable to download webpage: HTTP Error 429: Too Many Requests (caused by HTTPError()); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
====================================
I am using a current updated version of Windows 10.
I  am running youtube-dl version 2020.07.28
I've checked that all provided URLs are alive and playable in a browser
I've checked that all URLs and arguments with special characters are properly quoted or escaped

None of the videos are age restricted

**Comment by penyuan** [comment]
> Edit: Got it working with the browser plugin and the `--cookies` option.

This only works sporadically for me, very often I still get the "Unable to extract video data" error. I'm not sure what triggers the error.

**Comment by mixon2020** [comment]
Thank you will try that

**Comment by penyuan** [comment]
> Thank you will try that

Well like I said it only works for me occasionally. Hopefully a more fundamental solution can be found for the "unable to extract video data" error.

**Comment by Cyrix126** [comment]
I don't know if it's helpful, but if you're not connected to an account on youtube, you can still see a restricted video by changing in the url "watch?v=" with "embed". Unfortunately doing this with youtube-dl doesn't work.

**Comment by tezJR** [comment]
Is there an easy way for those without programming experience to apply the patch locally? And how do I create/find the cookies.txt file?

**Comment by omgitsraven** [comment]
If you're running it in Python, there's no programming necessary, just change this line in this file: https://github.com/ytdl-org/youtube-dl/pull/26100/files

If you're running it from an .exe file (as recommended on Windows), I'm not sure what to recommend, I haven't looked into the compilation process myself

**Comment by tezJR** [comment]
I'm running it from the .exe in Windows 10. I have Python 3.7.4 but wouldn't know how to use it for this without step-by-step instructions. If someone wishes to walk me through that, my email's in my profile. If not, thank you for your time and I'll wait for it to be merged.

**Comment by nicolaasjan** [comment]
> Is there an easy way for those without programming experience to apply the patch locally? And how do I create/find the cookies.txt file?

A `cookies.txt` file can be created with the "cookies.txt" extension.
Do this while being logged in to YouTube.

For Firefox:
https://addons.mozilla.org/en-US/firefox/addon/cookies-txt/
For Chrome:
https://chrome.google.com/webstore/detail/cookiestxt/njabckikapfpffapmjgojcnbfjonfjfg

Then run youtube-dl like this:
`youtube-dl.exe --cookies C:\Users\tezJR\Downloads\cookies-txt https://www.youtube.com/watch?v=$VIDEO-ID`
(adjust that command to where your `cookies.txt` file is actually located and replace `$VIDEO-ID` with  the ID of your restricted video).

**Comment by tezJR** [comment]
That worked! Thank you very much, nicolaasjan!

**Comment by nicolaasjan** [comment]
> That worked! Thank you very much, nicolaasjan!

No problem.

I have sent you a mail with instructions to compile `youtube-dl.exe` yourself.
It's not that complicated when you have Python installed.

**Comment by Rikk** [comment]
What's happening with youtube-dl team? No commits in a long time, and no fix for this obvious bug that has been reported many times?

**Comment by upex** [comment]
> I ask because I tried two approaches to provide cookies for a logged-in user and both didn't work (providing text file and --add-header option).
> I assume I got something wrong in the format then...
> Edit: Got it working with the browser plugin and the `--cookies` option. And now it downloads the video with an age-check.



> I ask because I tried two approaches to provide cookies for a logged-in user and both didn't work (providing text file and --add-header option).
> I assume I got something wrong in the format then...
> Edit: Got it working with the browser plugin and the `--cookies` option. And now it downloads the video with an age-check.

Hey @Xan-Kun , i did the same thing but every time I try to fetch URLs, the cookies.txt file gets override!! Is it expected behaviour? Also, I am running those commands in AWS hosted ubuntu machine. Please help.

**Comment by upex** [comment]
> > Is there an easy way for those without programming experience to apply the patch locally? And how do I create/find the cookies.txt file?
> 
> A `cookies.txt` file can be created with the "cookies.txt" extension.
> Do this while being logged in to YouTube.
> 
> For Firefox:
> https://addons.mozilla.org/en-US/firefox/addon/cookies-txt/
> For Chrome:
> https://chrome.google.com/webstore/detail/cookiestxt/njabckikapfpffapmjgojcnbfjonfjfg
> 
> Then run youtube-dl like this:
> `youtube-dl.exe --cookies C:\Users\tezJR\Downloads\cookies-txt https://www.youtube.com/watch?v=$VIDEO-ID`
> (adjust that command to where your `cookies.txt` file is actually located and replace `$VIDEO-ID` with the ID of your restricted video).

Hey @nicolaasjan , i did the same thing but every time I try to fetch URLs, the cookies.txt file gets override!! Is it expected behaviour? Also, I am running those commands in AWS hosted ubuntu machine. Please help.

**Comment by nicolaasjan** [comment]
> > > Is there an easy way for those without programming experience to apply the patch locally? And how do I create/find the cookies.txt file?
> > 
> > 
> > A `cookies.txt` file can be created with the "cookies.txt" extension.
> > Do this while being logged in to YouTube.
> > For Firefox:
> > https://addons.mozilla.org/en-US/firefox/addon/cookies-txt/
> > For Chrome:
> > https://chrome.google.com/webstore/detail/cookiestxt/njabckikapfpffapmjgojcnbfjonfjfg
> > Then run youtube-dl like this:
> > `youtube-dl.exe --cookies C:\Users\tezJR\Downloads\cookies-txt https://www.youtube.com/watch?v=$VIDEO-ID`
> > (adjust that command to where your `cookies.txt` file is actually located and replace `$VIDEO-ID` with the ID of your restricted video).
> 
> Hey @nicolaasjan , i did the same thing but every time I try to fetch URLs, the cookies.txt file gets override!! Is it expected behaviour? Also, I am running those commands in AWS hosted ubuntu machine. Please help.

I read somewhere, that youtube-dl changes the expiration date of the cookies in it, so that it keeps working as intended.
In the header of the file it says:
`# This file is generated by youtube-dl.  Do not edit.`

The expiration dates are in Unix Timestamp format, so `1603788661` means `10/27/2020 @ 8:51am (UTC)`
You can convert these values online here:
https://www.unixtimestamp.com/index.php

**Comment by spiralofhope** [comment]
> @nicolaasjan said:
> I read somewhere, that youtube-dl changes the expiration date of the cookies in it, so that it keeps working as intended.

I tested and confirmed it's being very heavily rewritten.

**Comment by ayunami2000** [comment]
This worked for me (via [reddit](https://www.reddit.com/r/youtubedl/comments/hs66fa/dailymotion_error_unable_to_download_video_data/)):
add `-f http-1080-0` to the arguments, like `youtube-dl -f http-1080-0 <url>`
(I didn't even have to use cookies!)

**Comment by Xan-Kun** [comment]
> Hey @Xan-Kun , i did the same thing but every time I try to fetch URLs, the cookies.txt file gets override!! Is it expected behaviour? Also, I am running those commands in AWS hosted ubuntu machine. Please help.

@upex Hi there. Yes it does! I just checked and didn't realise it before, but when I start the download, the cookies file gets overwritten.

**Comment by ayunami2000** [comment]
Perhaps try to specify quality? That worked for me but I'm not
entirely sure if I ever actually tested it with no args. See if that
works for you.

**Comment by windroldev** [comment]
**To me:** removing YouTube-dl downloaded using `sudo apt install youtube-dl` and installing it using `pip3 install youtube-dl` solved the issue!

**Comment by walkoncross** [comment]
> **To me:** removing YouTube-dl downloaded using `sudo apt install youtube-dl` and installing it using `pip3 install youtube-dl` solved the issue!
> 
> **FOR LINUX USERS ONLY!!!**

This works on MacOS for me. 
"""
$ brew uninstall youtube-dl
Uninstalling /usr/local/Cellar/youtube-dl/2020.11.1.1... (2,203 files, 16.4MB)

$ pip3 install youtube-dl

Collecting youtube-dl
  Downloading youtube_dl-2020.12.12-py2.py3-none-any.whl (1.8 MB)
     |████████████████████████████████| 1.8 MB 1.1 MB/s
Installing collected packages: youtube-dl
Successfully installed youtube-dl-2020.12.12
"""

**Comment by Laraibmazz** [comment]
Try installing it with snap instead if apt or apt-get

**sudo snap install youtube-dl**

**Comment by SatyamGaba** [comment]
> **To me:** removing YouTube-dl downloaded using `sudo apt install youtube-dl` and installing it using `pip3 install youtube-dl` solved the issue!
> 
> **FOR LINUX USERS ONLY!!!**

I get following error:

> `-bash: /usr/bin/youtube-dl: No such file or directory`

Can anyone help me to resolve it?

**Comment by sfolje0** [comment]
> 
> 
> > **To me:** removing YouTube-dl downloaded using `sudo apt install youtube-dl` and installing it using `pip3 install youtube-dl` solved the issue!
> > **FOR LINUX USERS ONLY!!!**
> 
> I get following error:
> 
> > `-bash: /usr/bin/youtube-dl: No such file or directory`
> 
> Can anyone help me to resolve it?

Hi @SatyamGaba,

Did you try running `python3 -m youtube_dl` (i.e. running reinstalled python's modul `youtube-dl`)?
Note also: youtube_dl and not youtube-dl in the above command.

Explained:
I suspect that your specific problem is that when you uninstalled `youtube-dl` using `apt`, it also deleted the binary /usr/bin/youtube-dl and therefore comand `$ youtube-dl` in bash will return your-mentioned error. When you 'reinstalled' it with pip, it stayed inside python's (private?) packages without creating yotube-dl executable. But that's no problem, just run it via python as said.
